mmseg/models/losses/inverseform_loss.py

```diff
@@ -0,0 +1,177 @@
+    model_dict = model.state_dict()
+    updated_model_dict = {}
+    for k_model, v_model in model_dict.items():
+        if k_model.startswith('model') or k_model.startswith('module'):
+            k_updated = '.'.join(k_model.split('.')[1:])
+            updated_model_dict[k_updated] = k_model
+        else:
+            updated_model_dict[k_model] = k_model
+
+    updated_pretrained_dict = {}
+    for k, v in pretrained_dict.items():
+        if k.startswith('model') or k.startswith('modules'):
+            k = '.'.join(k.split('.')[1:])
+        if k in updated_model_dict.keys() and model_dict[k].shape == v.shape:
+            updated_pretrained_dict[updated_model_dict[k]] = v
+
+    model_dict.update(updated_pretrained_dict)
+    model.load_state_dict(model_dict)
+    return model
+
+
+@MODELS.register_module()
+class InverseFormLoss(nn.Module):
+    """InverseForm loss, https://arxiv.org/abs/2104.02745.
+
+    Args:
+        tile_factor(int): divide the image to several tile, Default 3.
+        resized_dim(int): resize input and output to
+            (resized_dim,2*resized_dim), Default 672.
+        inverseNet_path(str): the path of pretrained InverseNet, download from
+            https://github.com/Qualcomm-AI-research/InverseForm/releases/download/v1.0/distance_measures_regressor.pth
+        map_location(str): If you want to load pretrained model to cpu,
+            using `cpu`, Default None, using in gpu.
+        loss_name (str, optional): Name of the loss item. If you want this loss
+            item to be included into the backward graph, `loss_` must be the
+            prefix of the name. Defaults to 'loss_inverseform'.
+    """
+
+    def __init__(self,
+                 tile_factor: int = 3,
+                 resized_dim: int = 672,
+                 inverseNet_path:
+                 str = './checkpoints/distance_measures_regressor.pth',
+                 map_location: str = None,
+                 loss_name: str = 'loss_inverseform'):
+        super().__init__()
+        self.tile_factor = tile_factor
+        self.resized_dim = resized_dim
+        self.tiled_dim = self.resized_dim // self.tile_factor
+        self.INVERSEFORM_MODULE = inverseNet_path
+
+        inversenet_backbone = InverseNet()
+        self.inversenet = load_model_from_dict(inversenet_backbone,
+                                               self.INVERSEFORM_MODULE,
+                                               map_location)
+        for param in self.inversenet.parameters():
+            param.requires_grad = False
+
+    def forward(self, inputs: torch.tensor, targets: torch.tensor,
+                **kwargs) -> torch.Tensor:
+        """
+        Args:
+            inputs(torch.tensor): pred boundary binary map.
+            targets(torch.tensor): true boundary binary map.
+        Returns:
+            mean_square_inverse_loss(torch.tensor): InverseTransformNet loss.
+        """
+        assert inputs.shape == targets.shape
```

Comment:
The shape of inputs is `(N, C, H, W)` and the shape targets is `(N, H, W)`

Reply:
Should we convert the label to onehot?
https://github.com/Qualcomm-AI-research/InverseForm/blob/main/library/datasets/base_loader.py#L144